File: backend/app/schemas/document.py
from uuid import UUID

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: UUID
    project_id: UUID
    filename: str
    mime_type: str
    file_size: int
    status: str
    error_message: str | None = None
    chunk_count: int
    created_at: str

    model_config = {"from_attributes": True}


class DocumentUploadResponse(BaseModel):
    document: DocumentResponse
    message: str = "Document accepted for processing"

File: frontend/src/types/document.ts
export type DocumentStatus = 'processing' | 'ready' | 'failed'

export interface ProjectDocument {
  id: string
  project_id: string
  filename: string
  mime_type: string
  file_size: number
  status: DocumentStatus
  error_message: string | null
  chunk_count: number
  created_at: string
}

export interface DocumentUploadResponse {
  document: ProjectDocument
  message: string
}
